enos/utils/extra.py

```diff
@@ -389,3 +389,26 @@ def _build_indexes(resources):
             s.extend(list(indexes))
             all_indexes.append(s)
     return all_indexes
+
+
+def make_provider(env):
+    """Instantiates the provider.
+
+    Seeks into the configuration for the `provider` value. The value
+    SHOULD be, either a *string*, or a *dictionary with a `type` key*
+    that gives the provider name. Then used this value to instantiate
+    and return the provider.
+
+    """
+    provider_name = env['config']['provider']['type']\
```

Comment:
Is there a default value ? Should we have one ? or just throw an exception ? ot let the following lines throw an exception?